pkg/descheduler/controllers/migration/podbudgetfinder/pod_budget_finder.go

```diff
@@ -0,0 +1,119 @@
+package podbudgetfinder
+
+import (
+	"context"
+	"strconv"
+
+	policyv1alpha1 "github.com/openkruise/kruise-api/policy/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+	kubecontroller "k8s.io/kubernetes/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	"github.com/koordinator-sh/koordinator/pkg/descheduler/controllers/migration/controllerfinder"
+	"github.com/koordinator-sh/koordinator/pkg/util"
+	utilclient "github.com/koordinator-sh/koordinator/pkg/util/client"
+)
+
+type Interface interface {
+	GetPodsForPub(pub *policyv1alpha1.PodUnavailableBudget, active bool) ([]*corev1.Pod, int32, error)
+	GetPubForPod(pod *corev1.Pod) (*policyv1alpha1.PodUnavailableBudget, error)
+	GetExpectedScaleForPods(pods []*corev1.Pod) (int32, error)
+}
+
+type PodBudgetFinder struct {
```

Comment:
How does the code work if there is no PUB CRD in the cluster?

Reply:
filterMaxUnavailableByPUB will return false. ok I will make it true if no pub crd.